subsys/net/lib/shell/dhcpv4.c

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(net_shell);
+
+#include <stdint.h>
+#include <zephyr/net/dhcpv4_server.h>
+
+#include "net_shell_private.h"
+
+static int cmd_net_dhcpv4_server_start(const struct shell *sh, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_DHCPV4_SERVER)
+	struct net_if *iface = NULL;
+	struct in_addr base_addr;
+	int idx, ret;
+
+	idx = get_iface_idx(sh, argv[1]);
+	if (idx < 0) {
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (!iface) {
+		PR_WARNING("No such interface in index %d\n", idx);
+		return -ENOEXEC;
+	}
+
+	if (net_addr_pton(AF_INET, argv[2], &base_addr)) {
+		PR_ERROR("Invalid address: %s\n", argv[2]);
+		return -EINVAL;
+	}
+
+	ret = net_dhcpv4_server_start(iface, &base_addr);
+	if (ret == -EALREADY) {
+		PR_WARNING("DHCPv4 server already running on interface %d\n", idx);
+	} else if (ret < 0) {
+		PR_ERROR("DHCPv4 server failed to start on interface %d, error %d\n",
+			 idx, -ret);
+	} else {
+		PR("DHCPv4 server started on interface %d\n", idx);
+	}
+#else /* CONFIG_NET_DHCPV4_SERVER */
+	PR_INFO("Set CONFIG_NET_DHCPV4_SERVER to enable DHCPv4 server support.\n");
+#endif /* CONFIG_NET_DHCPV4_SERVER */
+	return 0;
+}
+
+static int cmd_net_dhcpv4_server_stop(const struct shell *sh, size_t argc, char *argv[])
+{
+#if defined(CONFIG_NET_DHCPV4_SERVER)
+	struct net_if *iface = NULL;
+	int idx, ret;
+
+	idx = get_iface_idx(sh, argv[1]);
+	if (idx < 0) {
+		return -ENOEXEC;
+	}
+
+	iface = net_if_get_by_index(idx);
+	if (!iface) {
+		PR_WARNING("No such interface in index %d\n", idx);
+		return -ENOEXEC;
+	}
+
+	ret = net_dhcpv4_server_stop(iface);
+	if (ret == -ENOENT) {
+		PR_WARNING("DHCPv4 server is not running on interface %d\n", idx);
+	} else if (ret < 0) {
+		PR_ERROR("DHCPv4 server failed to stop on interface %d, error %d\n",
+			 idx, -ret);
+	} else {
+		PR("DHCPv4 server stopped on interface %d\n", idx);
+	}
+#else /* CONFIG_NET_DHCPV4_SERVER */
+	PR_INFO("Set CONFIG_NET_DHCPV4_SERVER to enable DHCPv4 server support.\n");
```

Comment:
I suggest we follow the same pattern as in other net-shell files so save some memory.
So
```
	PR_INFO("Set %s to enable %s support.\n",
		"CONFIG_NET_DHCPV4_SERVER",
		"DHCPv4 server");
```
Same comment to the start command.

Reply:
Thanks, updated.